x-pack/metricbeat/module/stan/channels/data.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package channels
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstriface"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+var (
+	moduleSchema = s.Schema{
+		"server": s.Object{
+			"id": c.Str("server_id"),
+		},
+		"cluster": s.Object{
+			"id": c.Str("cluster_id"),
+		},
+	}
+	channelSchema = s.Schema{
+		"name":      c.Str("name"),
+		"messages":  c.Int("msgs"),
+		"bytes":     c.Int("bytes"),
+		"first_seq": c.Int("first_seq"),
+		"last_seq":  c.Int("last_seq"),
+		"depth":     c.Int("depth", s.Optional), // aggregated by the module
+	}
+)
+
+// Subscription stores subscription related information
+type Subscription struct {
+	QueueName    string `json:"queue_name"`
+	IsDurable    bool   `json:"is_durable"`
+	IsOffline    bool   `json:"is_offline"`
+	IsStalled    bool   `json:"is_stalled"`
+	PendingCount int64  `json:"pending_count"`
+	LastSent     int64  `json:"last_sent"`
+}
+
+// Channel stores channel related information
+type Channel struct {
+	Name          string         `json:"name"`
+	Msgs          int64          `json:"msgs"`
```

Comment:
Can this be extended to full word?

Reply:
I would prefer to keep it 1to1 mapping here and change it in the schema https://github.com/elastic/beats/pull/14839/files/4701ec6c00a521c12211c2090f65bdb8216e3a11#diff-fd89e13c3682b4a680252526b585b1d2R28